### ISSUE TITLE
[strict-boolean-expressions] `ignore-rhs` disables errors further down the line

### ISSUE THREAD

**Issue by OliverJAsh** [palantir/tslint#4875]
### Bug Report

- __TSLint version__: 5.20.0
- __TypeScript version__: 3.6.2
- __Running TSLint via__: CLI

#### Reproduction using [TSLint Playground](https://palantir.github.io/tslint-playground)

https://palantir.github.io/tslint-playground/?saved=N4Igxg9gJgpiBcJZgDYEMBOMAEkB2AzgC7YC2AngIIYZrnzbEYCWeA5gNoC6A3ADp5k6LLgiESFAEIQIKGGjwMARjLkL+efMWwwAHgAcsBAszHYAvGXLTZ8vNgBkDq9Vrke2APSfsAUQMwYEQMeBA6NBAYAHTYlEEArmgoIWEwERgCzABm2AAUeoYwxqZ4AJTYwAC+Xj7++oHB4RiRMXFEicnYoU2RIAA04GJZzGwIIMAC2Nh8IBjxcgQzDBP2U9MgTMxBALQqtgrbBUYmYosI2BxEczB96yOhWNsYABZnXJPYlQJfeCCVQA

For some reason this rule doesn't seem to error in the playground?

#### TypeScript code being linted

```ts
declare const myArray: string[];
declare const myBoolean: boolean;
const expression = myBoolean && myArray; // Expect: no error. Actual: no error
if (expression) {} // Expect: error. Actual: no error
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "strict-boolean-expressions": [true, "ignore-rhs"]
  }
}
```

Related: https://github.com/palantir/tslint/pull/4159

**Comment by OliverJAsh** [comment]
Actually this is something else: https://github.com/palantir/tslint/issues/4876

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
